fix(interactive-examples): improve a11y + fix unsupported icon

caugner opened the pull request:
<!-- 🙌 Thanks for contributing! Adding details below will help us to merge your PR faster. -->

### Description

Bundles a couple of fixes, mainly accessibility of interactive examples:

1. [fix(play): add aria-live to runner/console](https://github.com/mdn/fred/commit/f3f5982e54166dd114928f33e9cb6a2b7ef0a487) should cause screen readers to announce output/console changes (e.g. when clicking "Run")
2. [fix(interactive-example): make choices a list](https://github.com/mdn/fred/commit/8acad00906c49443fb4631089c7acbd206d5c758) should cause screen readers to announce the CSS interactive examples choices as a list
3. [fix(interactive-example): add aria-label to choices](https://github.com/mdn/fred/commit/2f352776531a7df8ad8cf5a20227138d33fb4565) should cause screen readers to announce the interactive examples as a group using their heading
4. [fix(interactive-example): disable Reset until choice update](https://github.com/mdn/fred/pull/542/commits/03b4151cc34a84c40cee856039a56bc153b81e40) should avoid some confusion for screen reader users
5. [fix(interactive-examples): announce unsupported value](https://github.com/mdn/fred/commit/ef28cd46d6e3b7b6d74a8e5e0f384a8dade0e312) should fix [this issue reported in yari](https://github.com/mdn/yari/issues/12776)
6. [fix(interactive-example): show alert icon if unsupported](https://github.com/mdn/fred/commit/dcde79fd3a4fee22dba096ce5144aef32c610fbd) fixes a regression I noticed while fixing the previous item

### Motivation

Improve accessibility of interactive examples, and fix the unsupported CSS property/value icon.

### Additional details

<!-- 🔗 Link to documentation, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

Fixes https://github.com/mdn/fred/issues/522.

caugner committed:
fix(play): add aria-live to runner/console

caugner committed:
fix(interactive-example): make choices a list

caugner committed:
fix(interactive-example): add aria-label to choices

caugner committed:
fix(play-editor): avoid update event on init

caugner committed:
fix(interactive-example): disable Reset until choice update

caugner committed:
fix(interactive-example): disable Reset until choice update

caugner committed:
fix(interactive-examples): announce unsupported value

caugner committed:
fix(interactive-example): show alert icon if unsupported

caugner commented on components/interactive-example/element.css:
```diff
@@ -273,25 +273,21 @@ mdn-ix-tab-wrapper {
-          background-size: 1rem;
-
-          opacity: 1;
+          mask-image: url("../icon/triangle-alert.svg");
```
Note: It looks like this had regressed in https://github.com/mdn/fred/commit/08a03380352cd09047fa964ac97ce1605a3b19b8.

caugner commented on components/interactive-example/with-choices.js:
```diff
@@ -104,23 +111,29 @@ export const InteractiveExampleWithChoices = (Base) =>
             class="choice-wrapper"
             @click=${this.#choiceFocus}
             @focus=${this.#choiceSelect}
             @update=${this.#choiceUpdate}
```
Address https://github.com/mdn/fred/pull/542/files/dcde79fd3a4fee22dba096ce5144aef32c610fbd#r2273743349

```suggestion
            @update=${this.#choiceUpdate}
            aria-label=${this.l10n`Value select`}
```

caugner committed:
fix(interactive-example): add aria-label to choice list

caugner commented on components/interactive-example/with-choices.js:
```diff
@@ -104,23 +111,29 @@ export const InteractiveExampleWithChoices = (Base) =>
             >
           </header>
-          <div
+          <ul
```
Resolved via 4e7fc4f0343f80b94b3686aff8c6df6d0c9313f1.

caugner committed:
Revert "fix(interactive-example): disable Reset until choice update"

This reverts commit 46c8ceb695458796f6b66f3cf2e55e0983f297a1.

caugner committed:
Revert "fix(play-editor): avoid update event on init"

This reverts commit d51da5d9ada52d6a44b6482cbe6ccabc7ad57432.

caugner committed:
fix(interactive-example): add aria-label to console/tabs examples

LeoMcA commented on components/interactive-example/with-choices.js:
```diff
@@ -133,11 +138,17 @@ export const InteractiveExampleWithChoices = (Base) =>
                     minimal="true"
                     .delay=${100}
                     .value=${code?.trim()}
+                    aria-label=${ifDefined(
+                      this.__choiceUnsupported[index]
+                        ? this
+                            .l10n`The current value is not supported by your browser.`
+                        : undefined,
+                    )}
```
I think it not having a label implicitly suggests it is supported - so I wouldn't add one.

I think auto selecting the option when focusing is nice: why add an extra step to select the option? But happy to see it changed if we receive specific feedback about it being bad.

LeoMcA commented on components/play-console/element.js:
```diff
@@ -114,7 +114,7 @@ export class MDNPlayConsole extends LitElement {
   render() {
     return html`
-      <ul>
+      <ul aria-live="polite">
```
Could `tabindex="0"` be hiding this change? If we had to choose, I'd go with the more "standards compliant" approach, which seems to be this one.

caugner replied:
Removed in 9ce0b38e1f022ac0bf00f37f0f0fcf2799ff0a16.

LeoMcA reviewed (approved):
Looks good to me, I'll leave it to you whether you want to remove `tabindex=0` and merge!

caugner committed:
Merge branch 'main' into 522-improve-example-a11y

caugner committed:
Merge branch '522-disable-reset-button-until-edit' into 522-improve-example-a11y

caugner committed:
chore(play-console): remove tabindex

The tabindex should no longer be necessary, now
that we have `aria-live="polite"` on an ancestor.

caugner commented:
Merging now, as the changes here, and the merged changes were both were already approved.
